src/category_theory/sites/sheaf.lean

```diff
@@ -0,0 +1,244 @@
+    { intros y hy,
+      ext x,
+      apply (hP S hS).is_separated_for.ext,
+      intros Y' f hf,
+      rw [presieve.is_sheaf_for.valid_glue _ _ _ hf, ← hy _ hf],
+      refl } }
+end
+
+/--
+The category of sheaves taking values in Type is the same as the category of set-valued sheaves.
+-/
+@[simps]
+def Sheaf_equiv_SheafOfTypes : Sheaf J (Type v) ≌ SheafOfTypes J :=
+{ functor :=
+  { obj := λ S, ⟨S.1, (Sheaf_iff_SheafOfTypes _ _).1 S.2⟩,
+    map := λ S₁ S₂ f, f },
+  inverse :=
+  { obj := λ S, ⟨S.1, (Sheaf_iff_SheafOfTypes _ _).2 S.2⟩,
+    map := λ S₁ S₂ f, f },
+  unit_iso := nat_iso.of_components (λ X, ⟨𝟙 _, 𝟙 _, by tidy, by tidy⟩) (by tidy),
+  counit_iso := nat_iso.of_components (λ X, ⟨𝟙 _, 𝟙 _, by tidy, by tidy⟩) (by tidy) }
+
+instance : inhabited (Sheaf (⊥ : grothendieck_topology C) (Type v)) :=
+⟨(Sheaf_equiv_SheafOfTypes _).inverse.obj (default _)⟩
+
+end category_theory
+
+namespace category_theory
+
+open opposite category_theory category limits sieve classical
+
+namespace presheaf
+
+-- under here is the equalizer story, which is equivalent if A has products (and doesn't
+-- make sense otherwise). It's described between 00VQ and 00VR in stacks.
+
+variables {C : Type v} [small_category C]
+variables {A : Type u} [category.{v} A] [has_products A]
+variables (J : grothendieck_topology C)
+variables {U : C} (R : presieve U)
+variables (P : Cᵒᵖ ⥤ A)
+
+def first_obj : A :=
+∏ (λ (f : Σ V, {f : V ⟶ U // R f}), P.obj (op f.1))
+
+variables [has_pullbacks C]
+
+/--
+The rightmost object of the fork diagram of https://stacks.math.columbia.edu/tag/00VM, which
+contains the data used to check a family of elements for a presieve is compatible.
+-/
+def second_obj : A :=
+∏ (λ (fg : (Σ V, {f : V ⟶ U // R f}) × (Σ W, {g : W ⟶ U // R g})),
+  P.obj (op (pullback fg.1.2.1 fg.2.2.1)))
+
+/-- The map `pr₀*` of https://stacks.math.columbia.edu/tag/00VL. -/
+def first_map : first_obj R P ⟶ second_obj R P :=
+pi.lift (λ fg, pi.π _ _ ≫ P.map pullback.fst.op)
+
+/-- The map `pr₁*` of https://stacks.math.columbia.edu/tag/00VL. -/
+def second_map : first_obj R P ⟶ second_obj R P :=
+pi.lift (λ fg, pi.π _ _ ≫ P.map pullback.snd.op)
+
+/--
+The left morphism of the fork diagram given in Equation (3) of [MM92], as well as the fork diagram
+of https://stacks.math.columbia.edu/tag/00VM.
+-/
+def fork_map : P.obj (op U) ⟶ first_obj R P :=
+pi.lift (λ f, P.map f.2.1.op)
+
+lemma w : fork_map R P ≫ first_map R P = fork_map R P ≫ second_map R P :=
+begin
+  apply limit.hom_ext,
+  rintro ⟨⟨Y, f, hf⟩, ⟨Z, g, hg⟩⟩,
+  simp only [first_map, second_map, fork_map, limit.lift_π, limit.lift_π_assoc, assoc,
+    fan.mk_π_app, subtype.coe_mk, subtype.val_eq_coe],
+  rw [← P.map_comp, ← op_comp, pullback.condition],
+  simp,
+end
+
+def is_sheaf' (P : Cᵒᵖ ⥤ A) : Prop := ∀ (U : C) (R : presieve U) (hR : generate R ∈ J U),
+nonempty (is_limit (fork.of_ι _ (w R P)))
+
+/-- Impl. An auxiliary lemma to convert between sheaf conditions. -/
```

Comment:
What does "Impl." mean?

Reply:
Impl is short for implementation - the idea is that this lemma isn't helpful on its own but it's only there as implementation to make the next one easier. I've expanded it out anyway!